sdk/web-pubsub/web-pubsub-client/src/utils/abortablePromise.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortError, AbortSignalLike } from "@azure/abort-controller";
+
+export async function abortablePromise<T>(
+  promise: Promise<T>,
+  signal: AbortSignalLike
+): Promise<T> {
+  if (signal.aborted) {
+    throw new AbortError();
```

Comment:
we usually use a standard message for this e.g.

```suggestion
    throw new AbortError("The operation was aborted.");
```